bindings/rust/src/lib.rs

```diff
@@ -746,6 +791,7 @@ impl<'a, D> Unicorn<'a, D> {
     }
 
     /// Add hook for x86 OUT instruction.
+    #[cfg(feature = "arch_x86")]
```

Comment:
But where can users define these features?

Reply:
That's already built-in the the features in `Cargo.toml`, by default they are all enabled (via the `arch_all` feature enabling all of the arch features). Right now only the `build.rs` uses it when compiling unicorn, this just extends it so we aren't compiling rust code for c code that already got filtered out

Reply:
https://github.com/unicorn-engine/unicorn/blob/dev/Cargo.toml#L43

Reply:
(That's why the tests still work, by default all architectures are enabled)

Reply:
Ohhhhhh, sry I was looking at master branch.